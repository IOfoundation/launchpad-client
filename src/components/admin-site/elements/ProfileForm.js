import React, {Fragment} from 'react';
import {Form, FieldArray} from 'formik';
import {PropTypes} from 'prop-types';
import {withStyles} from '@material-ui/core/styles';

import FormTextField from '@Shared/FormElements/TextFieldDefault';
import SelectElement from '@Shared/FormElements/Select';

import {
  sectionOnePhoto,
  sectionOneRegular,
  socialSection,
  phoneNumberSection,
} from './ProfileForm/elements';

const ProfileForm = props => {
  const {
    errors,
    handleBlur,
    handleChange,
    handleSubmit,
    touched,
    values,
    classes,
  } = props;

  return (
    <Form className="profile-form" onSubmit={handleSubmit}>
      <div className={classes.containerFlex}>
        <div className={classes.sectionOnePhotoElement}>
          {sectionOnePhoto.map(form => {
            return (
              <FormTextField
                key={form.key}
                error={touched[form.key] && Boolean(errors[form.key])}
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id={form.key}
                label={form.label}
                autocomplete={form.autocomplete}
                value={values[form.key]}
                helperText={form.helperText}
              />
            );
          })}
        </div>
        <div
          className={['img-wrapper', classes.sectionOnePhotoPhoto].join(' ')}
        >
          <img src="https://via.placeholder.com/150" />
        </div>
      </div>
      <div className={classes.sectionOneRegular}>
        {sectionOneRegular.map(form => {
          return (
            <div key={form.key} className={classes.sectionOneItem}>
              <FormTextField
                error={touched[form.key] && Boolean(errors[form.key])}
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id={form.key}
                label={form.label}
                multiline={form.multiline}
                autocomplete={form.autocomplete}
                value={values[form.key]}
                helperText={form.helperText}
              />
            </div>
          );
        })}
      </div>
      <div className={`${classes.card} m-top-16`}>
        <div className={classes.cardTitle}>
          <span className={`${classes.cardTitle}__media`}>
            {'Social Media Links'}
          </span>
          <span className={`${classes.cardTitle}__small`}>
            {'Please enter the full URL for social media sites.'}
          </span>
        </div>
        <div className={`${classes.cardContent}`}>
          {socialSection.map(form => {
            return (
              <FormTextField
                key={form.key}
                error={touched[form.key] && Boolean(errors[form.key])}
                errors={errors}
                handleBlur={handleBlur}
                handleChange={handleChange}
                id={form.key}
                label={form.label}
                autocomplete={form.autocomplete}
                value={values[form.key]}
                helperText={form.helperText}
              />
            );
          })}
        </div>
      </div>

      <div className={classes.card}>
        <div className={classes.cardTitle}>
          <span className={`${classes.cardTitle}__media`}>
            {'Phone Numbers'}
          </span>
          <span className={`${classes.cardTitle}__small`}>
            {
              'If the phone number belongs to a contact, please move it to the existing contact, or add a new contact.'
            }
          </span>
        </div>
        <div className={`${classes.cardContent} ${classes.phoneItemWrapper}`}>
          <FieldArray
            name="phones"
            render={arrayHelpers => (
              <Fragment>
                {values.phones.map((phone, index) => {
                  const phoneMap = Object.keys(phone).map((key, i) => {
                    const id = `phones[${index}].${key}`;
                    const map = phoneNumberSection[i];
                    const error =
                      touched.phones &&
                      touched.phones[index] &&
                      touched.phones[index][key] &&
                      Boolean(
                        errors.phones &&
                          errors.phones[index] &&
                          errors.phones[index][key]
                      );

                    return (
                      <div key={id} className={classes.phoneItem}>
                        {map.select ? (
                          <SelectElement
                            errors={errors}
                            field="phones"
                            handleBlur={handleBlur}
                            handleChange={handleChange}
                            id={id}
                            key={id}
                            label={map.label}
                            name={id}
                            value={phone[key]}
                          />
                        ) : (
                          <FormTextField
                            autocomplete={map.autocomplete}
                            error={error}
                            errors={errors}
                            field="phones"
                            handleBlur={handleBlur}
                            handleChange={handleChange}
                            helperText={map.helperText}
                            id={key}
                            key={id}
                            label={map.label}
                            name={id}
                            value={phone[key]}
                          />
                        )}
                      </div>
                    );
                  });

                  return (
                    // eslint-disable-next-line react/no-array-index-key
                    <Fragment key={index}>
                      {phoneMap}
                      <div
                        className={`${classes.btnWrapper} ${
                          classes.bottomLine
                        }`}
                      >
                        <button
                          type="button"
                          className={`btn btn__red ${classes.btn}`}
                          disabled={values.phones.length === 1}
                          onClick={() => arrayHelpers.remove(index)}
                        >
                          {'Delete Phone Number'}
                        </button>
                      </div>
                      <div className={classes.btnWrapper}>
                        <button
                          type="button"
                          className={`btn btn__submit ${classes.btn}`}
                          onClick={() =>
                            arrayHelpers.insert(index, {
                              phoneNumber: '1',
                              ext: '',
                              vanityNumber: '',
                              numberType: '',
                              department: '',
                              countryExt: '',
                            })
                          }
                        >
                          {'Add New Phone Number'}
                        </button>
                      </div>
                    </Fragment>
                  );
                })}
              </Fragment>
            )}
          />
        </div>
      </div>
    </Form>
  );
};

const styles = () => {
  return {
    containerFlex: {
      display: 'flex',
    },
    sectionOnePhotoElement: {
      flex: 1,
    },
    sectionOnePhotoPhoto: {
      height: '200px',
      width: '222px',
      border: '1px solid black',
      marginLeft: '24px',
    },
    sectionOneRegular: {
      display: 'flex',
      flexWrap: 'wrap',
    },
    sectionOneItem: {
      width: '100%',
      boxSizing: 'border-box',
      '&:nth-child(4)': {
        width: 'calc(33.3% - 24px)',
        marginRight: '24px',
      },
      '&:nth-child(5)': {
        width: 'calc(33.3% - 24px)',
        marginRight: '24px',
      },
      '&:nth-child(6)': {
        width: '33.3%',
      },
    },
    card: {
      border: '1px solid black',
      marginBottom: '40px',
    },
    cardTitle: {
      color: 'white',
      background: 'black',
      padding: '8px 12px',
      '&__media': {
        fontSize: '16px',
        lineHeight: '24px',
        marginRight: '8px',
      },
      '&__small': {
        fontSize: '12px',
        lineHeight: '16px',
        opacity: '0.75',
      },
    },
    cardContent: {
      padding: '16px',
      background: 'white',
    },
    phoneItemWrapper: {
      display: 'flex',
      flexWrap: 'wrap',
    },
    phoneItem: {
      width: 'calc(45% - 24px)',
      marginRight: '24px',
    },
    btnWrapper: {
      width: '100%',
    },
    btn: {
      width: '260px',
      margin: '15px 0',
    },
    bottomLine: {
      borderBottom: '1px solid #E0E0E0',
    },
  };
};

ProfileForm.propTypes = {
  classes: PropTypes.shape({
    btn: PropTypes.string,
    btnWrapper: PropTypes.string,
    bottomLine: PropTypes.string,
    card: PropTypes.string,
    cardContent: PropTypes.string,
    cardTitle: PropTypes.string,
    containerFlex: PropTypes.string,
    phoneItem: PropTypes.string,
    phoneItemWrapper: PropTypes.string,
    sectionOnePhotoElement: PropTypes.string,
    sectionOnePhotoPhoto: PropTypes.string,
  }),
  errors: PropTypes.shape({
    email: PropTypes.string,
    password: PropTypes.string,
  }),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  isSubmitting: PropTypes.bool,
  isValid: PropTypes.bool,
  touched: PropTypes.shape({
    password: PropTypes.bool,
    contactEmail: PropTypes.bool,
    organizationName: PropTypes.bool,
    name: PropTypes.bool,
    website: PropTypes.bool,
    description: PropTypes.bool,
    accreditations: PropTypes.bool,
    dateIncorporation: PropTypes.bool,
    legalStatus: PropTypes.bool,
    fundingSources: PropTypes.bool,
    licenses: PropTypes.bool,
    taxIdentifier: PropTypes.bool,
    taxStatus: PropTypes.bool,
    twitter: PropTypes.bool,
    facebook: PropTypes.bool,
    linkedin: PropTypes.bool,
    phoneNumber: PropTypes.bool,
    ext: PropTypes.bool,
    vanityNumber: PropTypes.bool,
    numberType: PropTypes.bool,
    department: PropTypes.bool,
    countryExt: PropTypes.bool,
  }),
  values: PropTypes.shape({
    password: PropTypes.string,
    contactEmail: PropTypes.string,
    organizationName: PropTypes.string,
    name: PropTypes.string,
    website: PropTypes.string,
    description: PropTypes.string,
    accreditations: PropTypes.string,
    dateIncorporation: PropTypes.string,
    legalStatus: PropTypes.string,
    fundingSources: PropTypes.string,
    licenses: PropTypes.string,
    taxIdentifier: PropTypes.string,
    taxStatus: PropTypes.string,
    twitter: PropTypes.string,
    facebook: PropTypes.string,
    linkedin: PropTypes.string,
    phoneNumber: PropTypes.string,
    ext: PropTypes.string,
    vanityNumber: PropTypes.string,
    numberType: PropTypes.string,
    department: PropTypes.string,
    countryExt: PropTypes.string,
  }),
};

export default withStyles(styles)(ProfileForm);
