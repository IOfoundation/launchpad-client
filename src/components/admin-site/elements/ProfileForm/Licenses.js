import React from 'react';
import Grid from '@material-ui/core/Grid';
import {PropTypes} from 'prop-types';

import FormTextField from '@Shared/FormElements/TextFieldDefault';

const Licenses = props => {
  const {touched, errors, handleBlur, handleChange, values} = props;

  return (
    <Grid container={true} spacing={16}>
      <Grid item={true} xs={12}>
        <FormTextField
          key="licenses"
          error={touched.licenses && Boolean(errors.licenses)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="licenses"
          label="Licenses"
          autocomplete="off"
          value={values.licenses}
          type="text"
          helperText="You can enter multiple terms in this box by pressing the comma key after each one."
        />
      </Grid>
      <Grid item={true} xs={12}>
        <FormTextField
          key="taxIdentifier"
          error={touched.taxIdentifier && Boolean(errors.taxIdentifier)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="taxIdentifier"
          label="Tax Identifier"
          autocomplete="off"
          value={values.taxIdentifier}
          type="text"
          helperText="Tax identifier such as the Federal Employer Identification Number."
        />
      </Grid>
      <Grid item={true} xs={12}>
        <FormTextField
          key="taxStatus"
          error={touched.taxStatus && Boolean(errors.taxStatus)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="taxStatus"
          label="Tax Status"
          autocomplete="off"
          value={values.taxStatus}
          type="text"
          helperText="Internal Revenue Service tax designation, such as 501(c)(3)."
        />
      </Grid>
    </Grid>
  );
};

Licenses.propTypes = {
  errors: PropTypes.shape({}),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  touched: PropTypes.shape({}),
  values: PropTypes.shape({}),
};

export default Licenses;
