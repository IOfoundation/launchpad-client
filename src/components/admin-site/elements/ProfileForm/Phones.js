import React, {PureComponent, Fragment} from 'react';
import Grid from '@material-ui/core/Grid';
import {PropTypes} from 'prop-types';
import {FieldArray} from 'formik';
import {withStyles} from '@material-ui/core/styles';

import FormTextField from '@Shared/FormElements/TextFieldDefault';
import SelectElement from '@Shared/FormElements/Select';

import {numberType} from '@StaticData/data';

class Phones extends PureComponent {
  constructor(props) {
    super(props);
    this.displayErrors = this._errorsContext();
  }

  _errorsContext() {
    const {touched, errors} = this.props;
    return (index, key) => {
      return (
        touched.phones &&
        touched.phones[index] &&
        touched.phones[index][key] &&
        Boolean(
          errors.phones && errors.phones[index] && errors.phones[index][key]
        )
      );
    };
  }

  _detelePhone(arrayHelpers, index, id) {
    const {values} = this.props;

    if (id) {
      values.deletedPhones.push({
        id,
        _destroy: true,
      });
    }
    arrayHelpers.remove(index);
  }

  render() {
    const {handleBlur, handleChange, values, classes, errors} = this.props;

    return (
      <Grid container={true} spacing={16}>
        <FieldArray
          name="phones"
          render={arrayHelpers => (
            <Fragment>
              {values.phones.length > 0 ? (
                values.phones.map((phone, index) => {
                  const phoneMap = (
                    <Fragment>
                      <Grid item={true} xs={12} md={6}>
                        <FormTextField
                          autocomplete={'off'}
                          error={this.displayErrors(index, 'phoneNumber')}
                          errors={errors}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={'phoneNumber'}
                          label={'Number'}
                          name={`phones[${index}].phoneNumber`}
                          value={phone.phoneNumber}
                        />
                      </Grid>
                      <Grid item={true} xs={12} md={6}>
                        <FormTextField
                          autocomplete={'off'}
                          errors={errors}
                          error={this.displayErrors(index, 'ext')}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={'ext'}
                          label={'Extension'}
                          name={`phones[${index}].ext`}
                          value={phone.ext}
                        />
                      </Grid>
                      <Grid item={true} xs={12} md={6}>
                        <FormTextField
                          autocomplete={'off'}
                          errors={errors}
                          error={this.displayErrors(index, 'vanityNumber')}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={'vanityNumber'}
                          label={'Vanity Number (ex: 650-123-HELP)'}
                          name={`phones[${index}].vanityNumber`}
                          value={phone.vanityNumber}
                        />
                      </Grid>

                      <Grid item={true} xs={12} md={6}>
                        <SelectElement
                          errors={errors}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={`phones[${index}].numberType`}
                          label={'Number Type'}
                          name={`phones[${index}].numberType`}
                          value={phone.numberType}
                          selectOptions={numberType}
                        />
                      </Grid>

                      <Grid item={true} xs={12} md={6}>
                        <FormTextField
                          autocomplete={'off'}
                          errors={errors}
                          error={this.displayErrors(index, 'department')}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={'department'}
                          label={'Department'}
                          name={`phones[${index}].department`}
                          value={phone.department}
                        />
                      </Grid>

                      <Grid item={true} xs={12} md={6}>
                        <FormTextField
                          autocomplete={'off'}
                          errors={errors}
                          error={this.displayErrors(index, 'countryExt')}
                          field="phones"
                          handleBlur={handleBlur}
                          handleChange={handleChange}
                          id={'countryExt'}
                          label={'Country Prefix Code (for example: 1)'}
                          name={`phones[${index}].countryExt`}
                          value={phone.countryExt}
                        />
                      </Grid>
                    </Fragment>
                  );

                  return (
                    // eslint-disable-next-line react/no-array-index-key
                    <Fragment key={index}>
                      {phoneMap}
                      <Grid item={true} xs={12}>
                        <div className={classes.bottomLine}>
                          <button
                            type="button"
                            className={`btn btn__red ${classes.btn}`}
                            onClick={() =>
                              this._detelePhone(arrayHelpers, index, phone.id)
                            }
                          >
                            {'Delete Phone Number'}
                          </button>
                        </div>
                      </Grid>
                      <Grid item={true} xs={12}>
                        <button
                          type="button"
                          className={`btn btn__submit ${classes.btn}`}
                          onClick={() =>
                            arrayHelpers.insert(index, {
                              phoneNumber: '',
                              ext: '',
                              vanityNumber: '',
                              numberType: '',
                              department: '',
                              countryExt: '',
                            })
                          }
                        >
                          {'Add New Phone Number'}
                        </button>
                      </Grid>
                    </Fragment>
                  );
                })
              ) : (
                <div className={classes.center}>
                  <button
                    type="button"
                    className={`btn btn__submit ${classes.btn}`}
                    onClick={() =>
                      arrayHelpers.push({
                        phoneNumber: '',
                        ext: '',
                        vanityNumber: '',
                        numberType: '',
                        department: '',
                        countryExt: '',
                      })
                    }
                  >
                    {'Add New Phone Number'}
                  </button>
                </div>
              )}
            </Fragment>
          )}
        />
      </Grid>
    );
  }
}

const styles = theme => {
  return {
    btn: {
      width: '260px',
      margin: '0 0 16px',
      [theme.breakpoints.down('sm')]: {
        maxWidth: '243px',
        width: '100%',
      },
    },
    center: {
      alignContent: 'center',
      alignItems: 'center',
      display: 'flex',
      padding: '25px 8px 8px',
    },
    bottomLine: {
      borderBottom: '1px solid #E0E0E0',
    },
  };
};

Phones.propTypes = {
  classes: PropTypes.shape({
    bottomLine: PropTypes.string,
    btn: PropTypes.string,
    center: PropTypes.string,
  }),
  errors: PropTypes.shape({
    phones: PropTypes.arrayOf(PropTypes.shape({})),
  }),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  touched: PropTypes.shape({
    phones: PropTypes.arrayOf(PropTypes.shape({})),
  }),
  values: PropTypes.shape({
    phones: PropTypes.arrayOf(
      PropTypes.shape({
        phoneNumber: PropTypes.string,
        ext: PropTypes.string,
        vanityNumber: PropTypes.string,
        numberType: PropTypes.string,
        department: PropTypes.string,
        countryExt: PropTypes.string,
      })
    ),
  }),
};

export default withStyles(styles)(Phones);
