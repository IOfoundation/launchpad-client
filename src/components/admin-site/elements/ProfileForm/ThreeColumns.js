import React from 'react';
import Grid from '@material-ui/core/Grid';
import {PropTypes} from 'prop-types';

import FormTextField from '@Shared/FormElements/TextFieldDefault';

const ThreeColumns = props => {
  const {touched, errors, handleBlur, handleChange, values} = props;

  return (
    <Grid container={true} spacing={16}>
      <Grid item={true} xs={4}>
        <FormTextField
          key="dateIncorporation"
          error={touched.dateIncorporation && Boolean(errors.dateIncorporation)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="dateIncorporation"
          label="Date of Incorporation"
          autocomplete="off"
          value={values.dateIncorporation}
          type="date"
        />
      </Grid>
      <Grid item={true} xs={4}>
        <FormTextField
          key="legalStatus"
          error={touched.legalStatus && Boolean(errors.legalStatus)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="legalStatus"
          label="Legal Status"
          autocomplete="off"
          value={values.legalStatus}
          type="text"
        />
      </Grid>
      <Grid item={true} xs={4}>
        <FormTextField
          key="fundingSources"
          error={touched.fundingSources && Boolean(errors.fundingSources)}
          errors={errors}
          handleBlur={handleBlur}
          handleChange={handleChange}
          id="fundingSources"
          label="Funding Sources"
          autocomplete="off"
          value={values.fundingSources}
          type="text"
        />
      </Grid>
    </Grid>
  );
};

ThreeColumns.propTypes = {
  errors: PropTypes.shape({}),
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
  touched: PropTypes.shape({}),
  values: PropTypes.shape({}),
};

export default ThreeColumns;
