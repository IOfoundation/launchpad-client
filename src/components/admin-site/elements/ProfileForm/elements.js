export const sectionOnePhoto = [
  {
    key: 'name',
    label: 'Name',
    autocomplete: 'off',
  },
  {
    key: 'organizationName',
    label: 'Organization Display Name',
    autocomplete: 'off',
    helperText: 'This is how the name will appear to other users.',
  },
  {
    key: 'website',
    label: 'Organization Website',
    autocomplete: 'off',
  },
];

export const sectionOneRegular = [
  {
    key: 'description',
    label: 'Description',
    autocomplete: 'off',
    multiline: true,
  },
  {
    key: 'contactEmail',
    label: 'General Contact Email',
    autocomplete: 'off',
  },
  {
    key: 'accreditations',
    label: 'Accreditations',
    autocomplete: 'off',
    helperText:
      'You can enter multiple terms in this box by pressing the comma key after each one.',
  },
  {
    key: 'dateIncorporation',
    label: 'Date of Incorporation',
    autocomplete: 'off',
  },
  {
    key: 'legalStatus',
    label: 'Legal Status',
    autocomplete: 'off',
  },
  {
    key: 'fundingSources',
    label: 'Funding Sources',
    autocomplete: 'off',
  },
  {
    key: 'licenses',
    label: 'Licenses',
    autocomplete: 'off',
    helperText:
      'You can enter multiple terms in this box by pressing the comma key after each one.',
  },
  {
    key: 'taxIdentifier',
    label: 'Tax Identifier',
    autocomplete: 'off',
    helperText:
      'Tax identifier such as the Federal Employer Identification Number.',
  },
  {
    key: 'taxStatus',
    label: 'Tax Status',
    autocomplete: 'off',
    helperText: 'Internal Revenue Service tax designation, such as 501(c)(3).',
  },
];

export const socialSection = [
  {
    key: 'twitter',
    label: 'Twitter URL',
    autocomplete: 'off',
  },
  {
    key: 'facebook',
    label: 'Facebook URL',
    autocomplete: 'off',
  },
  {
    key: 'linkedin',
    label: 'Linkedin URL',
    autocomplete: 'off',
  },
];

export const phoneNumberSection = [
  {
    key: 'phoneNumber',
    label: 'Number',
    autocomplete: 'off',
  },
  {
    key: 'ext',
    label: 'Extension',
    autocomplete: 'off',
  },
  {
    key: 'vanityNumber',
    label: 'Vanity Number (ex: 650-123-HELP)',
    autocomplete: 'off',
  },
  {
    key: 'numberType',
    label: 'Number Type',
    autocomplete: 'off',
    select: true,
  },
  {
    key: 'department',
    label: 'Department',
    autocomplete: 'off',
  },
  {
    key: 'countryExt',
    label: 'Country Prefix Code (for example: 1)',
    autocomplete: 'off',
  },
];
