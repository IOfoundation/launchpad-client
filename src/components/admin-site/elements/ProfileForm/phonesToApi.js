import {sortArrayBy, falsyToString} from '@Utils';

export const getPhonesToApi = phones => {
  if (phones && phones.length > 0) {
    const sorted = sortArrayBy(phones, 'id');

    return sorted.map(phone => ({
      id: phone.id,
      department: falsyToString(phone.department),
      ext: falsyToString(phone.extension),
      numberType: falsyToString(phone.number_type),
      phoneNumber: falsyToString(phone.number),
      vanityNumber: falsyToString(phone.vanity_number),
      countryExt: falsyToString(phone.country_prefix),
    }));
  }
  return [];
};
