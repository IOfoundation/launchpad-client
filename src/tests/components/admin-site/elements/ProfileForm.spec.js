import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ProfileForm from '../../../../components/admin-site/elements/ProfileForm';

const props = {
  errors: {},
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  touched: {},
  values: {
    password: '',
    contactEmail: '',
    organizationName: '',
    name: '',
    website: '',
    description: '',
    accreditations: '',
    dateIncorporation: '',
    legalStatus: '',
    fundingSources: '',
    licenses: '',
    taxIdentifier: '',
    taxStatus: '',
    twitter: '',
    facebook: '',
    linkedin: '',
    phones: [
      {
        phoneNumber: '',
        ext: '',
        vanityNumber: '',
        numberType: '',
        department: '',
        countryExt: '',
      },
    ],
    images: {},
  },
  formik: {
    registerField: jest.fn(),
  },
};

describe('Admin-site <ProfileForm />', () => {
  it('renders snapshot of ProfileForm', () => {
    const wrapper = shallow(<ProfileForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
