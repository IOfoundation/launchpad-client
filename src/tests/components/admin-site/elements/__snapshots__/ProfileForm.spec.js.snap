// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin-site <ProfileForm /> renders snapshot of ProfileForm 1`] = `
<ProfileForm
  classes={
    Object {
      "bottomLine": "ProfileForm-bottomLine-13",
      "btn": "ProfileForm-btn-12",
      "btnWrapper": "ProfileForm-btnWrapper-11",
      "card": "ProfileForm-card-6",
      "cardContent": "ProfileForm-cardContent-8",
      "cardTitle": "ProfileForm-cardTitle-7",
      "containerFlex": "ProfileForm-containerFlex-1",
      "imageHeight": "ProfileForm-imageHeight-14",
      "phoneItem": "ProfileForm-phoneItem-10",
      "phoneItemWrapper": "ProfileForm-phoneItemWrapper-9",
      "sectionOneItem": "ProfileForm-sectionOneItem-5",
      "sectionOnePhotoElement": "ProfileForm-sectionOnePhotoElement-2",
      "sectionOnePhotoPhoto": "ProfileForm-sectionOnePhotoPhoto-3",
      "sectionOneRegular": "ProfileForm-sectionOneRegular-4",
    }
  }
  errors={Object {}}
  formik={
    Object {
      "registerField": [MockFunction],
    }
  }
  handleBlur={[MockFunction]}
  handleChange={[MockFunction]}
  handleSubmit={[MockFunction]}
  touched={Object {}}
  values={
    Object {
      "accreditations": "",
      "contactEmail": "",
      "dateIncorporation": "",
      "description": "",
      "facebook": "",
      "fundingSources": "",
      "images": Object {},
      "legalStatus": "",
      "licenses": "",
      "linkedin": "",
      "name": "",
      "organizationName": "",
      "password": "",
      "phones": Array [
        Object {
          "countryExt": "",
          "department": "",
          "ext": "",
          "numberType": "",
          "phoneNumber": "",
          "vanityNumber": "",
        },
      ],
      "taxIdentifier": "",
      "taxStatus": "",
      "twitter": "",
      "website": "",
    }
  }
/>
`;
